apps/web/src/pages/Applications/profileStep/profileStepValidation.tsx

```diff
@@ -13,11 +17,13 @@ import {
 const stepHasError = (
   applicant: Applicant,
   poolAdvertisement: PoolAdvertisement,
+  application: Omit<PoolCandidate, "pool">, // exists to type match hasError to stepHasError
+  isIAP?: boolean,
 ) => {
   const hasEmptyRequiredFields =
     aboutSectionHasEmptyRequiredFields(applicant) ||
     workLocationSectionHasEmptyRequiredFields(applicant) ||
-    diversityEquityInclusionSectionHasEmptyRequiredFields(applicant) ||
+    diversityEquityInclusionSectionHasEmptyRequiredFields(applicant, isIAP) ||
```

Comment:
@petertgiles why was isIAPPoolAdvertisement in applicationUtils causing a dependency loop?

Reply:
I gave it a whirl and carved out the context variable in https://github.com/GCTC-NTGC/gc-digital-talent/pull/6579/commits/5370f35bc92ab5e32acf1c35a208606481adacf8
I don't want to rename that validator so as to preserve consistency, I also want to avoid having multiple validators for one slice of the profile.

Reply:
@tristan-orourke :confused: 
ApplicationContext imports isIAPPoolAdvertisement  from "~/utils/applicationUtils"
ApplicationUtils imports welcomeStepInfo from "~/pages/Applications/welcomeStep/welcomeStepInfo";
WelcomeStepInfo imports getPageInfo from "../ApplicationWelcomePage/ApplicationWelcomePage";
ApplicationWelcomePage imports  useApplicationContext  from "../ApplicationContext";

Looking at it now, I think the bad guy is the second one.  That function that builds the steps (including welcomeStepInfo) is probably way bigger than should be in a util file.

Reply:
If the data and functions for getStepInfo for all the steps were all in one common util file, would that resolve it?

Reply:
I don't think so - you would just get one cycle instead of five different ones (one per application step).  I think I would want to rename `applicationUtilts` to `applicationStepUtils` and start a new `applicationUtils` file with `isIAPPoolAdverstisement`.  However you slice it, the solution is to split the functions into different files, I think.

Reply:
@vd1992 you've side-stepped it by just manually doing the check in the validation function.  :smile: We'll have to do a bit of refactoring if we ever need to do a second check for DEI requirements but I think this is good for now.